c/shared/source/uforead/uforead.c

```diff
@@ -1813,10 +1951,72 @@ static bool parseFontInfoFDArray(ufoCtx h, xmlNodePtr cur) {
     return true;
 }
 
+/* set each GLIF_REC's iFD based on the FDArraySelect groups*/
+static bool parseFDArraySelectGroup(ufoCtx h, char* FDArraySelectKeyName, xmlNodePtr cur) {
+    GLIF_Rec *g;
+
+    if (!xmlStrEqual((cur)->name, (const xmlChar *) "array"))
+        return false;
+
+    char* FDIndex = memNew(h, sizeof(char) * 2);
+    strncpy(FDIndex, FDArraySelectKeyName + 14, 2);
+    if (FDIndex[1] == '.')
```

Comment:
Sadly, I'm pretty sure one can have many more than 10 elements in an FDArray. I'm pretty sure I've run into actual fonts with more than that.

One of the `strtol` variants is probably what you want here.

Reply:
Not sure why I even wrote it assuming it would have less than 10...should be fixed now + added a new test for roundtripping with 100 FDArrays